### ISSUE TITLE
Scope raises error when array is passed as argument

### ISSUE THREAD

**Issue by krzysiek1507** [rails/rails#25654]
### Steps to reproduce

``` ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'activerecord', '5.0.0'
  gem 'sqlite3'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
  end
end

class Post < ActiveRecord::Base
  has_many :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post

  scope :by_post_ids, -> (post_ids = []) { where(post_id: post_ids) }
end

class BugTest < Minitest::Test
  def test_array_as_argument_to_scope
    Comment.by_post_ids([1, 2, 3])
  end
end
```
### Expected behavior

It returns comments for `post_ids` passed as argument
### Actual behavior

Raise 

```
ArgumentError: wrong number of arguments (3 for 0..1)
```
### System configuration

**Rails version**: 5.0.0

**Ruby version**: 2.2.4

**Comment by alextaylor000** [comment]
I get the same behaviour when running the test case above.

I also ran it against activerecord 4.2.6 and the test case passes.

**Comment by alextaylor000** [comment]
Did a quick `git bisect` and it looks as if [this commit](5e0b555b453ea2ca36986c111512627d806101e7) introduced the regression. The array is getting destructured when executing the proc where previously it was not.

**Comment by rafaelfranca** [comment]
This was a bug in Ruby. It is fixed with ruby 2.2.6/2.3.2

**Comment by wspurgin** [comment]
Yikes, I wish I had found this sooner. I started making the same PR to fix a `wrong number of arguments (given 3, expected 0..1)` error. Any place in the upgrading guides for >5 where this could be mentioned?
Especially this bug means an upgrader to upgrade Ruby too?

I still wonder though if #25699 shouldn't be reconsidered? The [docs](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#ruby-versions) say this on the ruby version:

> ## 1.3 Ruby Versions
>Rails generally stays close to the latest released Ruby version when it's released:
>
>Rails 5 requires Ruby 2.2.2 or newer.

Since 2.2.2 is the base and 2.3.0~1 and 2.2.4 at least has this problem, shouldn't a fix be implemented to support those versions? Otherwise we should probably specify  a new minimum version required (or at least document this known issue for these version somewhere in the upgrade guide).
